lte/gateway/c/oai/tasks/nas/emm/sap/emm_cn.c

```diff
@@ -1289,6 +1294,94 @@ static int _emm_cn_cs_domain_mm_information_req(
   OAILOG_FUNC_RETURN(LOG_NAS_EMM, rc);
 }
 
+static int _emm_cn_pdn_disconnect_rsp(
+  emm_cn_pdn_disconnect_rsp_t *msg)
+{
+  OAILOG_FUNC_IN(LOG_NAS_EMM);
+  int rc = RETURNok;
+  uint32_t i = 0;
+  proc_tid_t pti;
+  #define ESM_SAP_BUFFER_SIZE 4096
+  uint8_t esm_sap_buffer[ESM_SAP_BUFFER_SIZE];
+  esm_cause_t esm_cause = ESM_CAUSE_SUCCESS;
+  bstring rsp = NULL;
+
+  ESM_msg esm_msg;
+  memset(&esm_msg, 0, sizeof(ESM_msg));
+
+  mme_app_desc_t *mme_app_desc_p = get_mme_nas_state(false);
+  ue_mm_context_t *ue_mm_context = mme_ue_context_exists_mme_ue_s1ap_id(
+    &mme_app_desc_p->mme_ue_contexts, msg->ue_id);
+
+
+  pti = ue_mm_context->emm_context.esm_ctx.esm_proc_data->pti;
+  /*
+   * Build the ESM message to be sent to UE
+   */
+  rc = esm_send_deactivate_eps_bearer_context_request(
+    pti,
+    msg->lbi,
+    &esm_msg.deactivate_eps_bearer_context_request,
+    ESM_CAUSE_REGULAR_DEACTIVATION);
+
+  /*
+   * Encode the ESM message
+   */
+  int size =
+    esm_msg_encode(&esm_msg, (uint8_t *) esm_sap_buffer, ESM_SAP_BUFFER_SIZE);
+
+  if (size > 0) {
+    rsp = blk2bstr(esm_sap_buffer, size);
+  }
+
+  /*
+   * Populate the bearer ID list to be sent in E-RAB Rel Cmd
+   * to enB
+   */
+  pdn_cid_t cid =
+    ue_mm_context->bearer_contexts[EBI_TO_INDEX(msg->lbi)]->pdn_cx_id;
+  pdn_context_t *pdn_context = ue_mm_context->pdn_contexts[cid];
+
+  ue_mm_context->emm_context.esm_ctx.bearers_to_be_rel =
+    calloc(1, ((pdn_context->esm_data.n_bearers)*(sizeof(ebi_t))));
+  //Fill the default bearer ID in 0 index
+  ue_mm_context->emm_context.esm_ctx.bearers_to_be_rel[0] =
+    pdn_context->default_ebi;
+  for (i = 0; i < pdn_context->esm_data.n_bearers; i++) {
+    int idx = ue_mm_context->pdn_contexts[cid]->bearer_contexts[i];
+    ue_mm_context->emm_context.esm_ctx.bearers_to_be_rel[i+1] =
```

Comment:
This will go out of bound. Add a skip condition before this line that checks if this is a default bearer (in which case apply continue statement), instead of i+1, use a different index j and increment j only if current i is not a default bearer.